nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/avro/AvroRecordSetWriter.java

```diff
@@ -68,16 +76,40 @@
         .required(true)
         .build();
 
+    static final PropertyDescriptor ENCODER_POOL_SIZE = new Builder()
+        .name("encoder-pool-size")
+        .displayName("Encoder Pool Size")
+        .description("Avro Writers require the use of an Encoder. Creation of Encoders is expensive, but once created, they can be reused. This property controls the maximum number of Encoders that" +
+            " can be pooled and reused. Setting this value too small can result in degraded performance, but setting it higher can result in more heap being used.")
```

Comment:
Just for clarification, I'd suggest adding a note mentioning that, this property doesn't have any effect with 'Embed Avro Schema' strategy.